### PR TITLE
SONARIAC-1341 Decouple `LocationShifter` from `KubernetesParser`

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/HelmInputFileContext.java [hunk 1]
```diff
@@ -81,4 +84,8 @@ public boolean isInChartRootDirectory() {
     var rootChartDirectory = HelmFileSystem.retrieveHelmProjectFolder(Path.of(inputFile.uri()), sensorContext.fileSystem());
     return inputFile.path().getParent() != null && inputFile.path().getParent().equals(rootChartDirectory);
   }
+
+  public LinesShifting sourceMap() {
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/HelmInputFileContext.java:88]: I'm not sure if `sourceMap` is a good name for the method and field, as the type is `LinesShifting`. 
And in Java a `Map` usually means a Map as collection.

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/LocationShifter.java [hunk 2]
```diff
@@ -242,16 +235,12 @@ public MarkedYamlEngineException shiftMarkedYamlException(InputFileContext input
...
-    return linesShiftingPerContext.computeIfAbsent(ctx.inputFile.uri(), context -> new LinesShifting());
-  }
-
   /**
    * Store information related to an original line number.
    * The {@link #linesData} Map contain the original line number as the key.
    * The original line length and target line number are stored in the value, as a {@link LineData} object.
    */
-  static class LinesShifting {
+  public static class LinesShifting {
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/LocationShifter.java:243]: Consider extracting this class and `LineData` to separate file as this file is quite big and `LinesShifting` is used in `HelmInputFileContext`

**Review by mstachniuk — APPROVED** [review]
LGTM and please see my comments

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitor.java [hunk 3]
```diff
@@ -67,7 +65,7 @@ public AdjustableContextAdapter(RuleKey ruleKey) {
     @Override
     public <T extends Tree> void register(Class<T> cls, BiConsumer<CheckContext, T> visitor) {
       AdjustableChecksVisitor.this.register(cls, statistics.time(ruleKey.rule(), (InputFileContext ctx, T tree) -> {
-        currentCtx = ctx;
+        currentCtx = (HelmInputFileContext) ctx;
```
[the remark this answers is not in the document]
Comment (nils-werner-sonarsource) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitor.java:68]: Yeah, this is tricky without building a completely new interface

**Pull request merged**